### PR TITLE
fix(android): Fix setting an active tab for TabGroup before opening.

### PR DESCRIPTION
**Pull request by ypbnv**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26843

**Description:**
Fixes setting an active tab for TabGroup before it is opened. Fixes backwards compatibility for
events - one unnecessary "focus" is currently triggered on TabGroup opening. Removes unused code. Clean up imports. Adds new unit tests. Updates docs.

**Note:** I have tested with both supported styles - default and bottom navigation. What I wonder is what is the best way to include the bottom navigation style in unit tests - add all the tests with it ( that would work, but for any new style possibly added later it may become too much work) or we use the style property randomly chosen for each test. With the latter we still may miss some edge cases. 

I will try to figure out the failing of the "blur" unit tests since this PR changes event firing a little bit.

**Test case:**
_app.js_
```js
var tabGroup = Ti.UI.createTabGroup();
 
var win1 = Titanium.UI.createWindow({
    title: 'Tab 1',
    backgroundColor: '#fff'
});
var tab1 = Titanium.UI.createTab({
title: 'Tab 1',
    window: win1
});
 
var label1 = Titanium.UI.createLabel({
    color: '#999',
    text: 'I am Window 1',
    textAlign: 'center',
    width: 'auto'
});
 
win1.add(label1);
var win2 = Titanium.UI.createWindow({
    title: 'Tab 2',
    backgroundColor: '#fff'
});
var tab2 = Titanium.UI.createTab({
    title: 'Tab 2',
    window: win2
});
 
var label2 = Titanium.UI.createLabel({
    color: '#999',
    text: 'I am Window 2',
    textAlign: 'center',
    width: 'auto'
});
 
win2.add(label2);
tabGroup.addTab(tab1);
tabGroup.addTab(tab2);
tabGroup.activeTab = 1;
Ti.API.info("### activeTab: " + tabGroup.activeTab);
tabGroup.open();

```

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java [hunk 1]
```diff
@@ -256,10 +248,12 @@ public void removeTabItemFromController(int position)
...
-		((TabGroupProxy) getProxy())
-			.getTabList()
-			.get(currentlySelectedIndex)
-			.fireEvent(TiC.EVENT_UNSELECTED, null, false);
+		if (currentlySelectedIndex != -1) {
+			((TabGroupProxy) getProxy())
+				.getTabList()
+				.get(currentlySelectedIndex)
+				.fireEvent(TiC.EVENT_UNSELECTED, null, false);
+		}
```
Comment (jquick-axway) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java]: We should check if `getProxy()` returns `null` in the `selectTabItemInController()` method and the `onMenuItemClick()` method, which can happen when the proxy has been released (ie: it's JavaScript object was garbage collected).

**Review by jquick-axway — CHANGES_REQUESTED** [review]

**Comment by jquick-axway** [comment]
Looks good @ypbnv. I'm just asking for a few more `null` checks just in case. Thanks.

**Comment by keerthi1032** [comment]
FR Passed. Active tab works fine .some tests are failed in Jenkins.waiting for that to get resolved. also we need Backport PR FOR  8.0.
Operating System
  Name                        = Mac OS X
  Version                     = 10.13.6
  Architecture                = 64bit
Node.js
  Node.js Version             = 8.12.0
  npm Version                 = 6.4.1
Titanium CLI
  CLI Version                 = 5.1.1
Titanium SDK
  SDK Version                 = 8.1.0.v20190301023310
Device   =samsung s5 android 6
Emulator    =google pixel 3 xL android 8

File: android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java [hunk 2]
```diff
@@ -445,14 +445,10 @@ protected void handlePostOpen()
 
 		TabProxy activeTab = handleGetActiveTab();
 		if (activeTab != null) {
-			selectedTab = activeTab;
+			selectedTab = null;
```
[the remark this answers is not in the document]
Comment (ypbnv) [reply android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java]: Are you referring to this `onTabSelected()` method?
https://github.com/appcelerator/titanium_mobile/blob/master/android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java#L529
If that is the case, it is intended for the `selectedTab` to be null in order to skip the sending of the `focus/blur` events when a tab is set as an active one before the opening of the TabGroup. But I may have not got what you mean.

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java (hunk 1 above)
Comment (ypbnv) [reply android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java]: Updated.

File: android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jquick-axway) [reply android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java]: Oh you're right. Never mind.

**Review by jquick-axway — APPROVED** [review]
CR: Pass

File: android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sgtcoolguy) [reply android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java]: Note that the unit test around blur events is now failing, and is likely related to the changes in this area...

[the remark this answers is not in the document]
Comment (ypbnv) [reply android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java]: After some digging I found that the "blur" unit test started failing because this PR fixed firing a pair of `focus/blur` events that are not supposed to be fired. That problem was introduced at the same time when the unit test was added (Both by me) and this is the reason it didn't fail every single time in other PRs. With my latest update this should be cleared up.

**Comment by ypbnv** [comment]
@jquick-axway Would you be able to take a look at my latest commit - I moved the nullifying of a Tab's parent - from when the views are released to when the Tab proxy itself is released. That is the reason the `blur` event was failing. Its source is the Tab and bubbles up to the TabGroup, but the focus change happens after the views of the TabGroup are released. And at the views release the parent was set to null, so in the end the event had nowhere to propagate to.

File: android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jquick-axway) [reply android/modules/ui/src/java/ti/modules/titanium/ui/TabGroupProxy.java]: The unit test is passing now.
